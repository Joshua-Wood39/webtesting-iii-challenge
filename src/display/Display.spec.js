import React from 'react';
import { render, fireEvent } from 'react-testing-library';
import renderer from 'react-test-renderer';
import 'react-testing-library/cleanup-after-each';
import Dashboard from '../dashboard/Dashboard.js';
import TestRenderer from 'react-test-renderer';


describe('Display Snap', () => {
    it('matches snap', () => {
        const tree = renderer.create(<Dashboard />).toJSON();
        expect(tree).toMatchSnapshot();
    })
})

describe('Gate Functions', () => {
    it('defaults to Open/Unlocked', () => {
        const { getByText } = render(<Dashboard />);

        getByText('Opened');
        getByText('Unlocked');
    })

    

    it('closes the gate', () => {
        const { getByText } = render(<Dashboard />);

        const button = getByText('Close Gate');
        fireEvent.click(button);

        getByText('Closed');
    })

    it('locks the gate', () => {
        const { getByText } = render(<Dashboard />);

        const closeButton = getByText('Close Gate');
        fireEvent.click(closeButton);
        const lockButton = getByText('Lock Gate');
        fireEvent.click(lockButton);

        getByText('Locked');
    })
    
    it('unlocks the gate', () => {
        const { getByText } = render(<Dashboard />);

        const closeButton = getByText('Close Gate');
        fireEvent.click(closeButton);
        const lockButton = getByText('Lock Gate');
        fireEvent.click(lockButton);
        const unlockButton = getByText('Unlock Gate');
        fireEvent.click(unlockButton);

        getByText('Unlocked');
    })

    it('opens the gate', () => {
        const { getByText } = render(<Dashboard />);

        const closeButton = getByText('Close Gate');
        fireEvent.click(closeButton);
        const openButton = getByText('Open Gate');
        fireEvent.click(openButton);

        getByText('Opened');
        
    })
})

