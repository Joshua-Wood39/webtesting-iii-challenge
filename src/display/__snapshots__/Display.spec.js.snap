// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Snap matches snap 1`] = `
Array [
  <div
    className="display panel"
  >
    <div
      className="led green-led"
    >
      Unlocked
    </div>
    <div
      className="led green-led"
    >
      Opened
    </div>
  </div>,
  <div
    className="controls panel"
  >
    <button
      className="toggle-btn"
      disabled={true}
      onClick={[Function]}
    >
      Lock Gate
    </button>
    <button
      className="toggle-btn"
      disabled={false}
      onClick={[Function]}
    >
      Close Gate
    </button>
  </div>,
]
`;
